mlrun/runtimes/function.py

```diff
@@ -1396,3 +1317,151 @@ def get_nuclio_deploy_status(
 
     text = "\n".join(outputs) if outputs else ""
     return state, address, name, last_log_timestamp, text, function_status
+
+
+def _compile_nuclio_archive_config(
+    nuclio_spec,
+    function: RemoteRuntime,
+    builder_env,
+    project=None,
+    auth_info=None,
+):
+    def get_secrets(keys):
+        secrets = {}
+        if (
+            project
+            and get_k8s_helper(silent=True).is_running_inside_kubernetes_cluster()
+        ):
+            secrets = function._get_k8s().get_project_secret_data(project, keys)
+        for key in keys:
+            if key in builder_env:
+                secrets[key] = builder_env.get(key)
+        return secrets
```

Comment:
No reason for this to be nested, it's making it harder to read the code
You don't really need `function._get_k8s()` you can simply `get_k8s_helper()` 
and you can pass it the rest of the variables